### PR TITLE
fix(app): use group in dynamic k8s client

### PR DESCRIPTION
**Pull request by olevski**
Some recent issues were possibly related to this.

It was never confirmed whether this would have helped or not. But it is still a good idea to use the API group if available to avoid confusion and problems in the future.

I checked the python k8s client and when the group is not passed in to the `resource.get` function it defaults to `None`

### CODE REVIEW

File: controller/utils.py [hunk 1]
```diff
@@ -189,20 +189,20 @@ def parse_du_command(capacity, bytes_multiplier=1024):
...
     Get the proper API for a certain resource. We cache the resources
     availabe in the cluster for 60 seconds in order to reduce the amount
     of unnecessary requests in busy clusters.
     """
     try:
-        return api_cache[(api_version, kind)]
+        return api_cache[(api_version, kind, group)]
     except KeyError:
         client = dynamic.DynamicClient(api_client.ApiClient())
         api_cache[(api_version, kind)] = client.resources.get(
```
Comment (ableuler) [inline controller/utils.py:202]: I think you're missing the group in the key here.

File: controller/utils.py [hunk 2]
```diff
@@ -189,20 +189,20 @@ def parse_du_command(capacity, bytes_multiplier=1024):
...
-            api_version=api_version, kind=kind
+            api_version=api_version, kind=kind, group=group,
```
Comment (ableuler) [inline controller/utils.py:203]: Does it still work for say pods with `group=None`? I'd assume this is the default set in the definition of `client.resources.get` anyway?

**Review by ableuler — CHANGES_REQUESTED** [review]

File: controller/utils.py (hunk 1 above)
Comment (olevski) [reply controller/utils.py:202]: Yeah. My bad.

**Review by ableuler — APPROVED** [review]
Definitely an improvement, even if it should not fix the problem we've seen. 👍

**Pull request merged**